stdlib/public/core/TypeLayoutConstraints.swift

```diff
@@ -0,0 +1,40 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// Define some compile-known protocols used to constraint type layout.
+// These protocols can be used e.g. in @_specialize attribute.
+
+// Type should be a trivial type. It's size should be 8 bits.
+public protocol _Trivial8 {
```

Comment:
Ping? It is not clear to me how these protocols are used. In principle we should need to represent these layout constraints with protocols since it will cause us to pass unnecessary witness tables, unless we do more work in IRGen to special case them (which is something I'd like to avoid, special behavior of AnyObject is enough of a hassle already)